Documentation/ceph-upgrade.md

```diff
@@ -181,6 +181,29 @@ kubectl -n $ROOK_SYSTEM_NAMESPACE get pods
 kubectl -n $ROOK_NAMESPACE get pods
 ```
 
+### Verify CephCluster Resource Health
+
+The upgrade health checks for Ceph and Rook may pass, but sanity checks will still prevent the upgrades from progressing if cluster components are not in good standing.
+For example, the MON health checks validate that the monitors are in quorum before allowing the upgrade, but also validate that there are an **odd number** of them in the config.
+
+To verify this, run the following command:
+
+```sh
+kubectl -n rook-ceph get cephclusters.ceph.rook.io 
+```
+
+In case the resource is not healthy (in this case the cluster has an even number MONs), or still `Progressing` you will see the following output:
+```console
+NAME        DATADIRHOSTPATH   MONCOUNT   AGE    PHASE   MESSAGE                        HEALTH
+rook-ceph   /var/lib/rook     4          492d   Progressing   failed to perform validation before cluster creation: mon count 4 cannot be even, must be odd to support a healthy quorum   HEALTH_WARN
```

Comment:
Due to a change in CI process starting in v1.6, updating docs on rook.io in older versions will be a bit manual.
For the upgrade to v1.3 was this issue with 4 mons the main issue that hit you? If so, what about keeping this PR simple by adding a single bullet to the "Before you Upgrade" section that mentions checking for even mons? Then  starting in 1.6 and newer docs we can add this full section on checking the cluster CR status. For the recent docs, we would need a PR against master.

Reply:
Every version is impacted by this until there is a check in place in the orchestrator to at least surface the error. 1.3-1.4-1.5 are for impacted at least, likely 1.6 too, not sure if the check has been added to 1.7.

The warning should be added to all of the doc versions as it's completely silent, and results in odd errors.

The reason I added the extended section instead of just a single bullet as it took me 3 months of digging to even realize that I should check for this, as the healthy state or the impact of a progressing change is not mentioned, hence it's not obvious.